packages/flutter_tools/test/general.shard/base/error_handling_file_system_test.dart

```diff
@@ -23,93 +21,170 @@ final Platform windowsPlatform = FakePlatform(
     MockFileSystem mockFileSystem;
     ErrorHandlingFileSystem fs;
 
     setUp(() {
       mockFileSystem = MockFileSystem();
-      fs = ErrorHandlingFileSystem(mockFileSystem);
+      fs = ErrorHandlingFileSystem(
+        delegate: mockFileSystem,
+        platform: windowsPlatform,
+      );
       when(mockFileSystem.path).thenReturn(MockPathContext());
-      testbed = Testbed(overrides: <Type, Generator>{
-        Platform: () => windowsPlatform,
-      });
     });
 
-    void writeTests({
-      String testName,
-      int errorCode,
-      String expectedMessage,
-    }) {
-      test(testName, () => testbed.run(() async {
-        final MockFile mockFile = MockFile();
-        when(mockFileSystem.file(any)).thenReturn(mockFile);
-        when(mockFile.writeAsBytes(
-          any,
-          mode: anyNamed('mode'),
-          flush: anyNamed('flush'),
-        )).thenAnswer((_) async {
-          throw FileSystemException('', '', OSError('', errorCode));
-        });
-        when(mockFile.writeAsString(
-          any,
-          mode: anyNamed('mode'),
-          encoding: anyNamed('encoding'),
-          flush: anyNamed('flush'),
-        )).thenAnswer((_) async {
-          throw FileSystemException('', '', OSError('', errorCode));
-        });
-        when(mockFile.writeAsBytesSync(
-          any,
-          mode: anyNamed('mode'),
-          flush: anyNamed('flush'),
-        )).thenThrow(FileSystemException('', '', OSError('', errorCode)));
-        when(mockFile.writeAsStringSync(
-          any,
-          mode: anyNamed('mode'),
-          encoding: anyNamed('encoding'),
-          flush: anyNamed('flush'),
-        )).thenThrow(FileSystemException('', '', OSError('', errorCode)));
-
-        final File file = fs.file('file');
-
-        expect(() async => await file.writeAsBytes(<int>[0]),
-               throwsToolExit(message: expectedMessage));
-        expect(() async => await file.writeAsString(''),
-               throwsToolExit(message: expectedMessage));
-        expect(() => file.writeAsBytesSync(<int>[0]),
-               throwsToolExit(message: expectedMessage));
-        expect(() => file.writeAsStringSync(''),
-               throwsToolExit(message: expectedMessage));
-      }));
-    }
-
-    writeTests(
-      testName: 'when writing to a full device',
-      errorCode: kDeviceFull,
-      expectedMessage: 'The target device is full',
-    );
-    writeTests(
-      testName: 'when the file is being used by another program',
-      errorCode: kUserMappedSectionOpened,
-      expectedMessage: 'The file is being used by another program',
-    );
+    testWithoutContext('when writing to a full device', () async {
+      writeTests(
+        mockFileSystem: mockFileSystem,
+        fs: fs,
+        errorCode: kDeviceFull,
+        expectedMessage: 'The target device is full',
+      );
+    });
+    testWithoutContext('when the file is being used by another program', () async {
+      writeTests(
```

Comment:
I think these tests would be easier to read if the expectations were still in the test bodies.

Reply:
Done